tiatoolbox/dataloader/wsireader.py

```diff
@@ -137,5 +144,166 @@ def read_region(self, start_w, start_h, end_w, end_h, level=0):
         im_region = openslide_obj.read_region(
             [start_w, start_h], level, [end_w - start_w, end_h - start_h]
         )
-        im_region = np.asarray(im_region)
+        im_region = np.asarray(im_region)[:, :, 0:3]
         return im_region
+
+    def slide_thumbnail(self):
+        """Read whole slide image thumbnail at 1.5x
+
+        Args:
+            self (WSIReader):
+
+        Returns:
+            ndarray : image array
+
+        Examples:
+            >>> from tiatoolbox.dataloader import wsireader
+            >>> wsi_obj = wsireader.WSIReader(input_dir="./",
+            ...     file_name="CMU-1.ndpi")
+            >>> slide_thumbnail = wsi_obj.slide_thumbnail()
+
+        """
+        openslide_obj = self.openslide_obj
+        tile_objective_value = 20
+
+        if self.objective_power == 0:
+            self.objective_power = np.int(
+                openslide_obj.properties[openslide.PROPERTY_NAME_OBJECTIVE_POWER]
+            )
+
+        rescale = np.int(self.objective_power / tile_objective_value)
+        slide_dimension = openslide_obj.level_dimensions[0]
+        slide_dimension_20x = np.array(slide_dimension) / rescale
+        thumb = openslide_obj.get_thumbnail(
+            (int(slide_dimension_20x[0] / 16), int(slide_dimension_20x[1] / 16))
+        )
+        thumb = np.asarray(thumb)
+
+        return thumb
+
+    def save_tiles(self):
+        """Generate JPEG tiles from whole slide images
```

Comment:
Might be nice to allow formats other than JPEG e.g. lossless ones such as PNG or TIFF to avoid re-encoding errors and faster loading of TIFF from disk for downstream analysis. Since cv2 is being used it should be as simple as changing the path extension.

Reply:
Thanks for the suggestion. Added this option in this commit https://github.com/TIA-Lab/tiatoolbox/commit/8e77fe848a9f94e771179b51e417e5642b416b37